packages/frontend-2/lib/workspaces/composables/management.ts

```diff
@@ -414,7 +414,25 @@ export const useWorkspaceUpdateRole = () => {
   const { triggerNotification } = useGlobalToast()
 
   return async (input: WorkspaceRoleUpdateInput) => {
-    const result = await mutate({ input }).catch(convertThrowIntoFetchResult)
+    const result = await mutate(
+      { input },
+      {
+        update: (cache) => {
+          if (!input.role) {
+            modifyObjectField(
+              cache,
+              getCacheId('Workspace', input.workspaceId),
+              'team',
+              () => {
+                cache.evict({
```

Comment:
You're mixing up 2 things here. cache.evict() works by itself, there's no need to wrap it within modifyObjectField - it will just evict the object w/ that ID from the cache.

You'd only need modifyObjectField(workspace.team), if you want to edit that specific field (a workspace's team array). In this case I don't think you need it, cause if you cache.evict() the collaborator beforehand, the reference is gonna dissapear from any arrays its in automatically.

Reply:
@andrewwallacespeckle Nvm, team isn't an array, its a collection. So yea, while it will disappear automatically from the collection's items array, you still need to update totalCount.

Reply:
Also - the field supports filters, so use the `autoEvictFiltered` option for modifyobjectfield